### PR TITLE
Adds device plugin allocation latency metric.

### PR DESCRIPTION
**Pull request by jiayingz**
For #53497

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/kubelet/cm/deviceplugin/manager.go [hunk 1]
```diff
@@ -171,6 +172,7 @@ func (m *ManagerImpl) Allocate(resourceName string, devs []string) (*pluginapi.A
 // Register registers a device plugin.
 func (m *ManagerImpl) Register(ctx context.Context, r *pluginapi.RegisterRequest) (*pluginapi.Empty, error) {
 	glog.Infof("Got registration request from device plugin with resource name %q", r.ResourceName)
+	metrics.DevicePluginRegistrationCount.WithLabelValues(r.ResourceName).Inc()
```
Comment (ScorpioCPH) [inline pkg/kubelet/cm/deviceplugin/manager.go]: Just wondering why not count the `successful` registration？

Comment (jiayingz) [reply pkg/kubelet/cm/deviceplugin/manager.go]: I think it is also useful to track failed ones. For now, we can start with a single metric to track both. If needs come, we can consider to add separate metrics for failed and successful registrations.

Comment (ScorpioCPH) [reply pkg/kubelet/cm/deviceplugin/manager.go]: lgtm

**Review by vishh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**